### PR TITLE
Implemented Draft/Publish Banner and Email

### PR DESCRIPTION
**Pull request by leobcgov**
# Fixes or Implements [AB#8219](https://qslvic.visualstudio.com/304a1f8c-dace-4f85-adf3-bf563d5b3a39/_workitems/edit/8219)

* [X] Enhancement
* [ ] Bug
* [ ] Documentation

## Description
- Draft should be the default status when creating a new Banner or Email.
- Expired Banner in "New" status should not be editted or deleted.
- All the fields (subject, desc., dates, priority) are readonly when editting a Banner / Email which is not in Draft.
image
image
image

## Testing

* [ ] Unit Tests Updated
* [ ] Functional Tests Updated
* [ ] Not Required

### Steps to Reproduce

If this is a bug, please provide details on how to reproduce the code.

### UI Changes

YES

### Browsers Tested

* [ ] Chrome
* [ ] Safari
* [ ] Edge
* [ ] Firefox

Provide an explanation for any test(s) not completed.

## Notes/Additional Comments

Any additional notes or comments that may be relevant.  Include any specialized deployment steps here.  

</br>
</br>
</br>
___
  
## Reviewer Guidelines

A reviewer should examine the PR if the Status above indicates the PR is **Ready**.
Once ready, the reviewer should read the information below as well as supporting information like the Azure ticket to ensure they have a good understanding of the request.

The reviewer should then:

1. Is the PR template complete and accurate?

    Confirm by reading all sections below.

1. Confirm the merge branch

    A normal feature/bug branch should merge into the Dev branch.  Exceptions for hotfix and sprint merges do occur and the target would then be Master.

1. Do the code commits have an appropriate comment and include linkage to the azure ticket by including a reference to "AB#nnnn"?

1. Review the Code Changes

    1. Do the changes match the description?

        The code commits should reflect the documented description and nothing more.  If additional changes are included the reviewer should call these out and request that the developer document the changes in Azure and in the description or remove the changed code.

    1. useful tips

        * Any obvious errors or omissions?
        * Does the code match the style of other similar work?
        * Is the code easy to understand?
        * Is the code well documented?
        * DB Migrations are appropriate?

If the Answer to any of the questions above is No, then a request for change should be the result of the review.

### CODE REVIEW

File: Apps/AdminWebClient/src/ClientApp/package.json [hunk 1]
```diff
@@ -12,6 +12,7 @@
     "@fortawesome/fontawesome-svg-core": "^1.2.30",
     "@fortawesome/free-solid-svg-icons": "^5.14.0",
     "@fortawesome/vue-fontawesome": "^2.0.0",
+    "bootstrap-vue": "^2.17.3",
```
Comment (tiagograf-AOT) [inline Apps/AdminWebClient/src/ClientApp/package.json:15]: why do we have bootstrap vue in our admin web client?

Comment (leobcgov) [reply Apps/AdminWebClient/src/ClientApp/package.json:15]: good catch, forgot to remove it as my 1st version used the toggle button.

File: Apps/AdminWebClient/src/ClientApp/src/components/core/modals/BannerModal.vue [hunk 2]
```diff
@@ -162,6 +183,20 @@ export default class BannerModal extends Vue {
     @Prop() editedItem!: Communication;
     @Prop() isNew!: number;
 
+    private get isDraft(): boolean {
+        if (
+            this.editedItem.communicationStatusCode == CommunicationStatus.Draft
```
Comment (FuriousLlama) [inline Apps/AdminWebClient/src/ClientApp/src/components/core/modals/BannerModal.vue:188]: return this.editedItem.communicationStatusCode === CommunicationStatus.Draft

File: Apps/AdminWebClient/src/ClientApp/src/components/core/modals/BannerModal.vue [hunk 3]
```diff
@@ -203,6 +238,15 @@ export default class BannerModal extends Vue {
         }
     }
 
+    private publishOrDraft() {
```
Comment (FuriousLlama) [inline Apps/AdminWebClient/src/ClientApp/src/components/core/modals/BannerModal.vue:241]: togglePusblist is more readable

File: Apps/AdminWebClient/src/ClientApp/src/components/core/modals/EmailModal.vue [hunk 4]
```diff
@@ -189,6 +213,29 @@ export default class EmailModal extends Vue {
...
+        } else {
+            return false;
+        }
+    }
+
+    private get publishingStatus(): string {
+        return this.isDraft ? "Publish" : "Draft";
+    }
+
+    private publishOrDraft() {
```
Comment (FuriousLlama) [inline Apps/AdminWebClient/src/ClientApp/src/components/core/modals/EmailModal.vue:230]: same as above

File: Apps/AdminWebClient/src/ClientApp/src/components/core/modals/EmailModal.vue [hunk 5]
```diff
@@ -189,6 +213,29 @@ export default class EmailModal extends Vue {
         }
     }
 
+    private get isDraft(): boolean {
+        if (
+            this.editedItem.communicationStatusCode == CommunicationStatus.Draft
```
Comment (FuriousLlama) [inline Apps/AdminWebClient/src/ClientApp/src/components/core/modals/EmailModal.vue:218]: same as above

File: Apps/AdminWebClient/src/ClientApp/src/components/core/modals/BannerModal.vue (hunk 3 above)
Comment (leobcgov) [reply Apps/AdminWebClient/src/ClientApp/src/components/core/modals/BannerModal.vue]: updated

File: Apps/AdminWebClient/src/ClientApp/src/components/core/modals/BannerModal.vue [hunk 6]
```diff
@@ -162,6 +183,21 @@ export default class BannerModal extends Vue {
     @Prop() editedItem!: Communication;
     @Prop() isNew!: number;
 
+    private get isDraft(): boolean {
```
Comment (FuriousLlama) [inline Apps/AdminWebClient/src/ClientApp/src/components/core/modals/BannerModal.vue:186]: I meant the whole function can be simplified as 
```
 private get isDraft(): boolean {
            return this.editedItem.communicationStatusCode === CommunicationStatus.Draft;
    }
```

Comment (leobcgov) [reply Apps/AdminWebClient/src/ClientApp/src/components/core/modals/BannerModal.vue:186]: ah how could I miss this :-)

**Review by tiagograf-AOT — APPROVED** [review]

**Review by FuriousLlama — APPROVED** [review]

**Pull request merged**